File: 0203.移除链表元素/0203-移除链表元素.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummy=ListNode(-1,head)
        ans=dummy
        while dummy.next:
            if dummy.next.val!=val:
                dummy=dummy.next
            else:
                dummy.next=dummy.next.next
        return ans.next

File: 0203.移除链表元素/0203-移除链表元素.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* removeElements(struct ListNode* head, int val){
    struct ListNode* dummyhead = (struct ListNode*)malloc(sizeof(struct ListNode));
    // dummyhead->val = 0;
    dummyhead->next = head;
    struct ListNode* cur = dummyhead;
    while(cur->next){
        if( cur->next->val != val){
            cur = cur->next;
        }else{
            struct ListNode* tmp = cur->next;
            cur->next = cur->next->next;
            free(tmp);
        }
    }
    head = dummyhead->next;
    free(dummyhead);
    return head;
}   